homeassistant/components/mochad.py

```diff
@@ -39,19 +39,21 @@ def setup(hass, config):
     conf = config[DOMAIN]
     host = conf.get(CONF_HOST)
     port = conf.get(CONF_PORT)
+    mqtt_client = hass.data[mqtt.DATA_MQTT]._mqttc
```

Comment:
Don't use the client. Just use the publish and/or subscribe functions that are available at the module level in the mqtt component.

Look at the mysensors component for an example of how to do this.

Reply:
well. the problem with that, I cannot make external pymochad_mqtt dependent on hass publish function. _mqttc is an instance of standard paho's mqtt client.

Reply:
Not specifically on home assistant publish function, but a publish function that publish to a mqtt broker. Why wouldn't that work?

Reply:
Do you mean pass paho's mqtt publish function as a pointer to pymochad_mqtt?

Reply:
No accept a callback for publishing in pymochad_mqtt with appropriate arguments. Build this callback in home assistant using the mqtt component publish function.

Reply:
We do the same in the mysensors component.

Reply:
Ok. Found the relevant code in mysensors. Basically it's a publish wrapper defined in hass side that is passed as a callback to sensor's driver. Reworking it accordingly. Should be able to test and commit later today.